{
	"id": "CVE-2024-38355",
	"tags": [
		{
			"label": "Product",
			"message": "socket.io"
		},
		{
			"label": "Version",
			"message": "= < 2.5.1 "
		},
		{
			"label": "Vulnerability",
			"message": "CWE-20: Improper Input Validation"
		},
		{
			"label": "Vulnerability",
			"message": "CWE-754: Improper Check for Unusual or Exceptional Conditions"
		}
	],
	"description": " Socket.IO is an open source, real-time, bidirectional, event-based, communication framework. A specially crafted Socket.IO packet can trigger an uncaught exception on the Socket.IO server, thus killing the Node.js process. This issue is fixed by commit `15af22fc22` which has been included in `socket.io@4.6.2` (released in May 2023). The fix was backported in the 2.x branch as well with commit `d30630ba10`. Users are advised to upgrade. Users unable to upgrade may attach a listener for the \"error\" event to catch these errors.",
	"poc": {
		"githubs": [
			"https://github.com/Y0ursTruly/Y0ursTruly"
		]
	}
}
